engine/storage_sql.go

```diff
@@ -115,20 +118,47 @@ func (self *SQLStorage) IsDBEmpty() (resp bool, err error) {
 	return true, nil
 }
 
+// update
 // Return a list with all TPids defined in the system, even if incomplete, isolated in some table.
-func (self *SQLStorage) GetTpIds() ([]string, error) {
-	rows, err := self.Db.Query(
-		fmt.Sprintf("(SELECT tpid FROM %s) UNION (SELECT tpid FROM %s) UNION (SELECT tpid FROM %s) UNION (SELECT tpid FROM %s) UNION (SELECT tpid FROM %s) UNION (SELECT tpid FROM %s)",
-			utils.TBLTPTimings,
-			utils.TBLTPDestinations,
-			utils.TBLTPRates,
-			utils.TBLTPDestinationRates,
-			utils.TBLTPRatingPlans,
-			utils.TBLTPRateProfiles))
-	if err != nil {
-		return nil, err
+func (self *SQLStorage) GetTpIds(colname string) ([]string, error) {
+	var rows *sql.Rows
+	var err error
+	if colname != "" {
```

Comment:
Here should not be needed to have if else. Use a string (ie: qryStr := ...) and add extra params to it if colName != ""